src/Semantics.js

```diff
@@ -196,6 +196,41 @@ Semantics.prototype.checkActionDicts = function() {
   }
 };
 
+Semantics.prototype.toRecipe = function() {
+  var sb = new common.StringBuffer();
```

Comment:
Beware of microbenchmarks...you sure the JIT didn't eliminate all the code between the `console.time()` and `console.timeEnd()`? ;-)